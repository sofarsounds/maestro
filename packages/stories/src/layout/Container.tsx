import React from 'react';
import styled from 'styled-components';
import { select } from '@storybook/addon-knobs';

import { Container } from 'uikit';

const ColContent = styled.div`
  background: #ccc;
  border-radius: 5px;
  text-align: center;
  padding: 10px;
`;

export default () => (
  <div>
    <Container
      size={select(
        'Size',
        ['default', 'small', 'large', 'fullscreen'],
        'default'
      )}
    >
      <ColContent>Container</ColContent>
    </Container>
  </div>
);
