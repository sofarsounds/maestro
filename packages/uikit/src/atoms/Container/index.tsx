import styled, { css } from 'styled-components';

type Sizes = 'default' | 'small' | 'large' | 'fullscreen';
interface ContainerProps {
  size?: string;
}

export default styled.div<ContainerProps>`
  ${({ theme, size }) => css`
    width: 100%;
    max-width: ${theme.dimensions.containerWidth[(size || 'default') as Sizes]};
    margin: 0 auto;
  `}
`;
