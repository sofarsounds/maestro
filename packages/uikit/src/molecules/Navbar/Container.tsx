import styled, { css } from '../../lib/styledComponents';

export default styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: 90%;
    display: flex;
    align-items: center;
    justify-content: space-between;

    ${theme.media.md`
        max-width: 85%;
      `}

    ${theme.media.lg`
        max-width: 80%;
      `}
  `}
`;
