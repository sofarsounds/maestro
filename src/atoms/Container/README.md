# Container

To implement a `Container` into your project you'll need to add this import
```js
import { Container } from '@sofarsounds/maestro'
```

After adding the import you can use the Container simply like this
```html
<Container>Content within the container</Container>
```

## Break Points

| Screen Size | Max Width |
| :---------- | :-------- |
| `< 540px`   | `100%`    |
| `>= 576px`  | `540px`   |
| `>= 768px`  | `720px`   |
| `>= 992px`  | `960px`   |
| `>= 1200px` | `1140px`  |

