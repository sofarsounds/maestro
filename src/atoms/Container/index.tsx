import styled, { css } from 'styled-components';

import { withFlex, WithFlexProps } from '../../util';

// TODO do we need different sizes per container?
// i.e. "small" or "large" container in addition to the default
// max-width?
interface ContainerProps extends WithFlexProps {
  size?: string;
}

export default styled.div<ContainerProps>`
  ${({ theme }) => css`
    width: 100%;
    margin: 0 auto;
    padding: 0 ${theme.ruler[4]}px;

    ${theme.media.sm`
      max-width: 540px;
      padding: 0 10px;
    `}

    ${theme.media.md`
      max-width: 740px;
    `}

    ${theme.media.lg`
      max-width:960px;
    `}

    ${theme.media.xl`
      max-width: 1160px;
    `}
  `}

  ${withFlex};
`;
