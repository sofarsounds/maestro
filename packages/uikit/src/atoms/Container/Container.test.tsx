import React from 'react';
import { mountWithTheme } from '../../test';

import Container from './index';

describe('Container', () => {
  it('renders the Container correctly with default width', () => {
    expect(mountWithTheme(<Container />)).toMatchSnapshot();
  });

  it('renders the Container correctly with small width', () => {
    expect(mountWithTheme(<Container size="small" />)).toMatchSnapshot();
  });

  it('renders the Container correctly with large width', () => {
    expect(mountWithTheme(<Container size="large" />)).toMatchSnapshot();
  });

  it('renders the Container correctly with fullscreen width', () => {
    expect(mountWithTheme(<Container size="fullscreen" />)).toMatchSnapshot();
  });
});
