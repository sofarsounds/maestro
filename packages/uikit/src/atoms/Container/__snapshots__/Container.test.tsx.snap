// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container renders the Container correctly with default width 1`] = `
.c0 {
  width: 100%;
  max-width: 800px;
  margin: 0 auto;
}

<ThemeProvider
  theme={
    Object {
      "borderRadius": Object {
        "button": "4px",
        "input": "2px",
      },
      "colours": Object {
        "backToBlack": "#111111",
        "black": "#000000",
        "blackBetty": "#1E1E1E",
        "blueSmoke": "#949494",
        "green100": "#CFEACC",
        "green200": "#9DDCAC",
        "green300": "#72D08A",
        "green400": "#4FC571",
        "green50": "#E6F7EA",
        "green500": "#23BA57",
        "green600": "#10AD52",
        "green700": "#039842",
        "green900": "#006721",
        "macyGrey": "#DBDBDB",
        "paintItBlack": "#333333",
        "primary": "#008736",
        "redRedWine": "#EF4423",
        "silverSprings": "#F5F5F5",
        "whiteDenim": "#FFFFFF",
      },
      "dimensions": Object {
        "buttonHeight": "44px",
        "containerWidth": Object {
          "default": "800px",
          "fullscreen": "100%",
          "large": "1000px",
          "small": "600px",
        },
        "inputHeight": "44px",
      },
      "fontSizes": Object {
        "base": "16px",
        "body1": "16px",
        "body2": "14px",
        "button": "14px",
        "caption": "10px",
        "h1": "96px",
        "h2": "60px",
        "h3": "48px",
        "h4": "34px",
        "h5": "24px",
        "h6": "20px",
        "overline": "10px",
        "subtitle1": "16px",
        "subtitle2": "14px",
        "title": "74px",
      },
      "fonts": Object {
        "regular": "Open Sans",
      },
      "media": Object {
        "lg": [Function],
        "md": [Function],
        "sm": [Function],
        "xl": [Function],
        "xs": [Function],
      },
      "ruler": Array [
        0,
        4,
        8,
        12,
        16,
        20,
        24,
        28,
        32,
        36,
        40,
        44,
        48,
        52,
        56,
        60,
        64,
        68,
        72,
        76,
        80,
      ],
      "utils": Object {
        "transition": [Function],
      },
    }
  }
>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  ",
              [Function],
              "
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="c0"
      />
    </StyledComponent>
  </styled.div>
</ThemeProvider>
`;

exports[`Container renders the Container correctly with fullscreen width 1`] = `
.c0 {
  width: 100%;
  max-width: 100%;
  margin: 0 auto;
}

<ThemeProvider
  theme={
    Object {
      "borderRadius": Object {
        "button": "4px",
        "input": "2px",
      },
      "colours": Object {
        "backToBlack": "#111111",
        "black": "#000000",
        "blackBetty": "#1E1E1E",
        "blueSmoke": "#949494",
        "green100": "#CFEACC",
        "green200": "#9DDCAC",
        "green300": "#72D08A",
        "green400": "#4FC571",
        "green50": "#E6F7EA",
        "green500": "#23BA57",
        "green600": "#10AD52",
        "green700": "#039842",
        "green900": "#006721",
        "macyGrey": "#DBDBDB",
        "paintItBlack": "#333333",
        "primary": "#008736",
        "redRedWine": "#EF4423",
        "silverSprings": "#F5F5F5",
        "whiteDenim": "#FFFFFF",
      },
      "dimensions": Object {
        "buttonHeight": "44px",
        "containerWidth": Object {
          "default": "800px",
          "fullscreen": "100%",
          "large": "1000px",
          "small": "600px",
        },
        "inputHeight": "44px",
      },
      "fontSizes": Object {
        "base": "16px",
        "body1": "16px",
        "body2": "14px",
        "button": "14px",
        "caption": "10px",
        "h1": "96px",
        "h2": "60px",
        "h3": "48px",
        "h4": "34px",
        "h5": "24px",
        "h6": "20px",
        "overline": "10px",
        "subtitle1": "16px",
        "subtitle2": "14px",
        "title": "74px",
      },
      "fonts": Object {
        "regular": "Open Sans",
      },
      "media": Object {
        "lg": [Function],
        "md": [Function],
        "sm": [Function],
        "xl": [Function],
        "xs": [Function],
      },
      "ruler": Array [
        0,
        4,
        8,
        12,
        16,
        20,
        24,
        28,
        32,
        36,
        40,
        44,
        48,
        52,
        56,
        60,
        64,
        68,
        72,
        76,
        80,
      ],
      "utils": Object {
        "transition": [Function],
      },
    }
  }
>
  <styled.div
    size="fullscreen"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  ",
              [Function],
              "
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      size="fullscreen"
    >
      <div
        className="c0"
        size="fullscreen"
      />
    </StyledComponent>
  </styled.div>
</ThemeProvider>
`;

exports[`Container renders the Container correctly with large width 1`] = `
.c0 {
  width: 100%;
  max-width: 1000px;
  margin: 0 auto;
}

<ThemeProvider
  theme={
    Object {
      "borderRadius": Object {
        "button": "4px",
        "input": "2px",
      },
      "colours": Object {
        "backToBlack": "#111111",
        "black": "#000000",
        "blackBetty": "#1E1E1E",
        "blueSmoke": "#949494",
        "green100": "#CFEACC",
        "green200": "#9DDCAC",
        "green300": "#72D08A",
        "green400": "#4FC571",
        "green50": "#E6F7EA",
        "green500": "#23BA57",
        "green600": "#10AD52",
        "green700": "#039842",
        "green900": "#006721",
        "macyGrey": "#DBDBDB",
        "paintItBlack": "#333333",
        "primary": "#008736",
        "redRedWine": "#EF4423",
        "silverSprings": "#F5F5F5",
        "whiteDenim": "#FFFFFF",
      },
      "dimensions": Object {
        "buttonHeight": "44px",
        "containerWidth": Object {
          "default": "800px",
          "fullscreen": "100%",
          "large": "1000px",
          "small": "600px",
        },
        "inputHeight": "44px",
      },
      "fontSizes": Object {
        "base": "16px",
        "body1": "16px",
        "body2": "14px",
        "button": "14px",
        "caption": "10px",
        "h1": "96px",
        "h2": "60px",
        "h3": "48px",
        "h4": "34px",
        "h5": "24px",
        "h6": "20px",
        "overline": "10px",
        "subtitle1": "16px",
        "subtitle2": "14px",
        "title": "74px",
      },
      "fonts": Object {
        "regular": "Open Sans",
      },
      "media": Object {
        "lg": [Function],
        "md": [Function],
        "sm": [Function],
        "xl": [Function],
        "xs": [Function],
      },
      "ruler": Array [
        0,
        4,
        8,
        12,
        16,
        20,
        24,
        28,
        32,
        36,
        40,
        44,
        48,
        52,
        56,
        60,
        64,
        68,
        72,
        76,
        80,
      ],
      "utils": Object {
        "transition": [Function],
      },
    }
  }
>
  <styled.div
    size="large"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  ",
              [Function],
              "
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      size="large"
    >
      <div
        className="c0"
        size="large"
      />
    </StyledComponent>
  </styled.div>
</ThemeProvider>
`;

exports[`Container renders the Container correctly with small width 1`] = `
.c0 {
  width: 100%;
  max-width: 600px;
  margin: 0 auto;
}

<ThemeProvider
  theme={
    Object {
      "borderRadius": Object {
        "button": "4px",
        "input": "2px",
      },
      "colours": Object {
        "backToBlack": "#111111",
        "black": "#000000",
        "blackBetty": "#1E1E1E",
        "blueSmoke": "#949494",
        "green100": "#CFEACC",
        "green200": "#9DDCAC",
        "green300": "#72D08A",
        "green400": "#4FC571",
        "green50": "#E6F7EA",
        "green500": "#23BA57",
        "green600": "#10AD52",
        "green700": "#039842",
        "green900": "#006721",
        "macyGrey": "#DBDBDB",
        "paintItBlack": "#333333",
        "primary": "#008736",
        "redRedWine": "#EF4423",
        "silverSprings": "#F5F5F5",
        "whiteDenim": "#FFFFFF",
      },
      "dimensions": Object {
        "buttonHeight": "44px",
        "containerWidth": Object {
          "default": "800px",
          "fullscreen": "100%",
          "large": "1000px",
          "small": "600px",
        },
        "inputHeight": "44px",
      },
      "fontSizes": Object {
        "base": "16px",
        "body1": "16px",
        "body2": "14px",
        "button": "14px",
        "caption": "10px",
        "h1": "96px",
        "h2": "60px",
        "h3": "48px",
        "h4": "34px",
        "h5": "24px",
        "h6": "20px",
        "overline": "10px",
        "subtitle1": "16px",
        "subtitle2": "14px",
        "title": "74px",
      },
      "fonts": Object {
        "regular": "Open Sans",
      },
      "media": Object {
        "lg": [Function],
        "md": [Function],
        "sm": [Function],
        "xl": [Function],
        "xs": [Function],
      },
      "ruler": Array [
        0,
        4,
        8,
        12,
        16,
        20,
        24,
        28,
        32,
        36,
        40,
        44,
        48,
        52,
        56,
        60,
        64,
        68,
        72,
        76,
        80,
      ],
      "utils": Object {
        "transition": [Function],
      },
    }
  }
>
  <styled.div
    size="small"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  ",
              [Function],
              "
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      size="small"
    >
      <div
        className="c0"
        size="small"
      />
    </StyledComponent>
  </styled.div>
</ThemeProvider>
`;
